lib/cli/src/commands/mod.rs

```diff
@@ -70,12 +71,21 @@ pub(crate) trait AsyncCliCommand: Send + Sync {
     type Output: Send + Sync;
 
     async fn run_async(self) -> Result<Self::Output, anyhow::Error>;
+
+    fn setup(&self) -> Result<(), anyhow::Error> {
+        ctrlc::set_handler(move || {
```

Comment:
nit: no need for the separate `crtlc` dependency, we can use `tokio::signal::ctrl_c` .

With the current structure it'll be easiest to just run it in a spawned task.